crates/driver/src/infra/solver/dto/auction.rs

```diff
@@ -68,31 +70,35 @@ impl Auction {
                 .iter()
                 .map(|order| {
                     let mut available = order.available(weth);
-                    // Solvers are unaware of the protocol fees. In case of volume based fees,
-                    // fee withheld by driver might be higher than the surplus of the solution. This
-                    // would lead to violating limit prices when driver tries to withhold the
-                    // volume based fee. To avoid this, we artifically adjust the order limit
-                    // amounts (make then worse) before sending to solvers, to force solvers to only
-                    // submit solutions with enough surplus to cover the fee.
+                    // In case of volume based fees, fee withheld by driver might be higher than the
+                    // surplus of the solution. This would lead to violating limit prices when
+                    // driver tries to withhold the volume based fee. To avoid this, we artificially
+                    // adjust the order limit amounts (make then worse) before sending to solvers,
+                    // to force solvers to only submit solutions with enough surplus to cover the
+                    // fee.
                     //
                     // https://github.com/cowprotocol/services/issues/2440
-                    if let Some(FeePolicy::Volume { factor }) = order.protocol_fees.first() {
-                        match order.side {
-                            Side::Buy => {
-                                // reduce sell amount by factor
-                                available.sell.amount = available
-                                    .sell
-                                    .amount
-                                    .apply_factor(1.0 / (1.0 + factor))
-                                    .unwrap_or_default();
-                            }
-                            Side::Sell => {
-                                // increase buy amount by factor
-                                available.buy.amount = available
-                                    .buy
-                                    .amount
-                                    .apply_factor(1.0 / (1.0 - factor))
-                                    .unwrap_or_default();
+                    if fee_handler == FeeHandler::Driver {
```

Comment:
Given that we now have a concept of a `FeeHandler` maybe we could actually have a struct for that that contains all the relevant code.
Then we would have a single place where all the fee related shenanigans happen. Most of it's functions would probably be no-ops for `FeeHandler::Solver`.
Not sure if this would actually end up nice and I think this idea should definitely not be explored in this PR (to avoid sneaking in bugs by accident) but maybe it could improve the code.

Reply:
@MartinquaXD this is a really good point. I didn't refactor it in this PR because I had plans to do it for this task: https://github.com/cowprotocol/services/issues/2461 if that is ok with you. In that task a new parameter enters the equation for the protocol fees, so it is a perfect task to refactor that in a way that it is clearer. WDYT?